kong/tools/utils.lua

```diff
@@ -61,11 +62,20 @@ local v4_uuid = uuid.generate_v4
 _M.uuid = uuid.generate_v4
 
 --- Seeds the random generator, use with care.
--- Kong already seeds this once per worker process. It's 
--- dangerous to ever call it again. So ask yourself
--- "Do I feel lucky?" Well, do ya, punk?
--- @function uuid_seed
-_M.uuid_seed = uuid.seed
+-- The uuid.seed() method will create a unique seed per worker
+-- process, using a combination of both time and the worker's pid.
+-- We then disable (or rather, no-op) `math.randomseed()` to prevent
+-- third-party modules from overriding our correct seed (many modules
+-- make a wrong usage of `math.randomseed()` by calling it multiple times
+-- or do not use unique seed for Nginx workers.
+function _M.randomseed()
+  local seed = uuid.seed()
+  ngx.log(ngx.DEBUG, "seeding random number generator for worker ",
+                      ngx.worker.id(), " with: ", seed)
+  -- luacheck: globals math
+  math.randomseed = function()end
```

Comment:
isn't this too late? it replaces it when seeding, so any other module might have cached the original function and still call it.

So store the original in an upvalue, and replace upon module loading? and then even make sure it is loaded first...